miqa/core/tests/pyppeteer/test_import_export.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+
+
+async def get_current_num_scans(page):
+    return len(await page.xpath('//ul[contains(@class, "scans")]/li'))
+
+
+async def edit_import_export_paths(page):
+    # Edit import path
+    import_path_input = await page.waitForXPath('//input[contains(@name, "import-path")]')
+    await import_path_input.click({'clickCount': 3})  # highlight the text to overwrite it
+    await import_path_input.type('samples/demo_project.csv')
+    await page.waitFor(1_000)
+
+    # Edit export path
+    import_path_input = await page.waitForXPath('//input[contains(@name, "export-path")]')
+    await import_path_input.click({'clickCount': 3})  # highlight the text to overwrite it
+    await import_path_input.type('samples/demo_project_export.csv')
+    await page.waitFor(1_000)
+
+    # Click save
+    await (await page.waitForXPath('//button[span[.=" Save "]][@type="submit"]')).click()
+    await page.waitFor(1_000)
+
+
+async def perform_import(page):
+    # Click import button, and again in the modal confirmation
+    await (await page.waitForXPath('//button[span[.=" Import "]]')).click()
+    await page.waitFor(1_000)
+    await (
+        await page.waitForXPath(
+            '//div[contains(@class, "v-card__actions")]//button[span[.=" Import "]]'
+        )
+    ).click()
+    await page.waitFor(6_000)
```

Comment:
Generally speaking, calling `waitFor(...)` immediately before `waitForXPath(...)` doesn't accomplish very much. [waitForXPath](https://puppeteer.github.io/puppeteer/docs/puppeteer.page.waitforxpath/) already waits for a while (30 seconds by default, I think? It's configurable when the `page` fixture is set up), and you can set an arbitrary timeout when calling waitForXPath. If it really takes more than 30s but less than 36s for the element to appear, I think it's probably better to just pass the `timeout` parameter to waitForXPath.

That said, there are exceptions when it does make sense to do exactly this. If you had a search widget with some search results already present, then you do a new search, you would need to wait a moment with `waitFor` for the old results to be cleared away, then wait for the new results with `waitForXPath`. This doesn't seem like one of those times, though.